source/MaterialXCore/Element.h

```diff
@@ -94,9 +94,13 @@ class Element : public enable_shared_from_this<Element>
     /// @name Name
     /// @{
 
-    /// Return the element's name string.  The name of a MaterialX element
-    /// must be unique among all elements at the same scope, and cannot be
-    /// modified after the element is created.
+    /// Set the element's name string.  The name of a MaterialX element must be
+    /// unique among all elements at the same scope.
+    /// @throws Exception if an element at the same scope already possesses the
+    ///    given name.
+    void setName(const string& name);
```

Comment:
That's a good suggestion, though I'd recommend waiting until we have a corresponding helper method (e.g. Document::updateNameReferences) that we can point the developer to.